lib/simplifyExpression/substituteScopeSearch/index.js

```diff
@@ -0,0 +1,54 @@
+const ChangeTypes = require('../../ChangeTypes');
+const Node = require('../../node');
+const TreeSearch = require('../../TreeSearch');
+
+// Searches through the tree, prioritizing deeper nodes, and substitutes
+// in-scope values for their respective expressions on a symbol node if possible.
+// Returns a Node.Status object.
+const search = TreeSearch.postOrder(scopeSubstitution);
+
+function scopeSubstitution(node, scope) {
+  if (Node.Type.isSymbol(node)) {
+    return substituteAndSimplifySymbols(node, scope);
+  }
+  else {
+    return Node.Status.noChange(node);
+  }
+}
+
+// SUBSTITUTES
+// Returns a Node.Status object with substeps
+function substituteAndSimplifySymbols(node, scope) {
+  if (!Node.Type.isSymbol(node)) {
+    return Node.Status.noChange(node);
+  }
+
+  let symbolName;
+  if (node.type === 'SymbolNode') {
```

Comment:
WAIT does this return true for unary minus too?

okay this makes sense https://github.com/socraticorg/mathsteps/blob/344d15b5d5e25251cef7d6976d6a73df16661d7d/lib/node/Type.js#L32

For any isSymbol thing you've added in this PR can you pass in `false` for the `allowUnaryMinus` argument? Then it can naturally recurse instead of you having to deal with it here. 

Sorry, that's weird that we did that (though I'm assuming there's a reason I've now forgotten XD but feel free to investigate if you're curious or open an issue)